common/src/main/java/io/netty/util/internal/svm/InternalLoggerFactorySubstitution.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal.svm;
+
+import io.netty.util.internal.logging.InternalLoggerFactory;
+import io.netty.util.internal.logging.JdkLoggerFactory;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/**
+ * Defaults to JDK logging in a native image instead of Slf4j and others.
+ */
+@TargetClass(className = "io.netty.util.internal.logging.InternalLoggerFactory")
+final class InternalLoggerFactorySubstitution {
+
+    private InternalLoggerFactorySubstitution() {
+    }
+
+    @Substitute
+    private static InternalLoggerFactory newDefaultFactory(String name) {
+        return JdkLoggerFactory.INSTANCE;
```

Comment:
should we do the same as in the original implementation and log that we use the JDK implementation ?

```java
InternalLoggerFactory  f = JdkLoggerFactory.INSTANCE;
f.newInstance(name).debug("Using java.util.logging as the default logging framework");
return f;
```

Reply:
Good idea as well…